sources/packages/backend/libs/services/src/health-check/health.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@nestjs/common";
+import { RedisOptions, Transport } from "@nestjs/microservices";
+import {
+  HealthIndicator,
+  HealthIndicatorResult,
+  HealthCheckError,
+  MicroserviceHealthIndicator,
+} from "@nestjs/terminus";
+import { ConfigService } from "@sims/utilities/config";
+import { ZeebeHealthIndicator } from "apps/workers/src/zeebe";
+
+@Injectable()
+export class HealthService extends HealthIndicator {
+  constructor(
+    private microservice: MicroserviceHealthIndicator,
+    private readonly configService: ConfigService,
+    private readonly zeebe: ZeebeHealthIndicator,
+  ) {
+    super();
+  }
+
+  /**
+   * Check the health of the application.
+   * @param key specifies if the health check is done for workers/queues.
+   * @returns status of the health check.
+   * In case of queues, we check if the redis is up and running.
+   * In case of workers, we check if the zeebe is ready to accept the connections.
+   */
+  async isHealthy(key: string): Promise<HealthIndicatorResult> {
+    let isHealthy;
+    let healthCheckResult;
+
+    if (key === "queues") {
+      const healthCheckResult = await this.checkRedisHealth();
+      isHealthy = healthCheckResult.redis.status === "up" ? true : false;
```

Comment:
could be `isHealthy  = healthCheckResult.redis.status === "up"` Sonar is complaining about the same